### PR TITLE
Make jobs permissions more strict: specific to a worker 

### PR DESCRIPTION
**Pull request by jinroh**
Followup of #950.

Improve the `GET /jobs/queue/:worker-type` route to include the queued and running jobs via a restricted permission on the worker.

This PR also includes some simplifications of the `pkg/jobs` package, mainly removing some unused abstraction for persistence layer.

### CODE REVIEW

File: web/data/permissions.go [hunk 1]
```diff
@@ -26,7 +26,6 @@ var blackList = map[string]bool{
 	consts.Files:            readable,
 	consts.Jobs:             readable,
 	consts.Notifications:    readable,
-	consts.Queues:           readable,
```
Comment (nono) [inline web/data/permissions.go]: Why this change?

File: web/data/permissions.go [hunk 2]
```diff
@@ -26,7 +26,6 @@ var blackList = map[string]bool{
 	consts.Files:            readable,
 	consts.Jobs:             readable,
```
[the remark this answers is not in the document]
Comment (jinroh) [reply web/data/permissions.go]: Indeed. Done in a335d0e119c1f5d5a9d6e14a900fe8cbd1b13077

File: web/data/permissions.go (hunk 1 above)
Comment (jinroh) [reply web/data/permissions.go]: I removed completely the `io.cozy.queues` doctype. It wasn't really required looking back.

**Comment by jinroh** [comment]
With these last to commits, the jobs permissions is not *actually* reduced by this PR to ease the update of applications that still depend on "whole-type" permissions on io.cozy.jobs. As soon as these applications are updated, we will revert these last two commits.
